pkg/util/cluster/aad.go

```diff
@@ -6,57 +6,44 @@ package cluster
 import (
 	"context"
 	"fmt"
-	"net/http"
 	"time"
 
-	azgraphrbac "github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
-	"github.com/Azure/go-autorest/autorest"
-	"github.com/Azure/go-autorest/autorest/date"
+	msgraph_apps "github.com/microsoftgraph/msgraph-sdk-go/applications"
+	msgraph_models "github.com/microsoftgraph/msgraph-sdk-go/models"
+	msgraph_errors "github.com/microsoftgraph/msgraph-sdk-go/models/odataerrors"
 	"k8s.io/apimachinery/pkg/util/wait"
-
-	"github.com/Azure/ARO-RP/pkg/util/uuid"
 )
 
-func (c *Cluster) getServicePrincipal(ctx context.Context, appID string) (string, error) {
-	// TODO: we are listing here rather than calling
-	// i.applications.GetServicePrincipalsIDByAppID() due to some missing
-	// permission with our dev/e2e applications
-	sps, err := c.serviceprincipals.List(ctx, fmt.Sprintf("appId eq '%s'", appID))
+func (c *Cluster) createApplication(ctx context.Context, displayName string) (string, string, error) {
+	appBody := msgraph_models.NewApplication()
+	appBody.SetDisplayName(&displayName)
+	appResult, err := c.spGraphClient.Applications().Post(ctx, appBody, nil)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
-	switch len(sps) {
-	case 0:
-		return "", nil
-	case 1:
-		return *sps[0].ObjectID, nil
-	default:
-		return "", fmt.Errorf("%d service principals found for appId %s", len(sps), appID)
-	}
-}
+	id := *appResult.GetId()
+	endDateTime := time.Now().AddDate(1, 0, 0)
 
-func (c *Cluster) createApplication(ctx context.Context, displayName string) (string, string, error) {
-	password := uuid.DefaultGenerator.Generate()
-
-	app, err := c.applications.Create(ctx, azgraphrbac.ApplicationCreateParameters{
-		DisplayName: &displayName,
-		PasswordCredentials: &[]azgraphrbac.PasswordCredential{
-			{
-				EndDate: &date.Time{Time: time.Now().AddDate(1, 0, 0)},
-				Value:   &password,
-			},
-		},
-	})
+	pwCredential := msgraph_models.NewPasswordCredential()
+	pwCredential.SetDisplayName(&displayName)
+	pwCredential.SetEndDateTime(&endDateTime)
+
+	pwCredentialRequestBody := msgraph_apps.NewItemAddPasswordPostRequestBody()
+	pwCredentialRequestBody.SetPasswordCredential(pwCredential)
+	// ByApplicationId is confusingly named, but it refers to
+	// the application's Object ID, not to the Application ID.
+	// https://learn.microsoft.com/en-us/graph/api/application-addpassword?view=graph-rest-1.0&tabs=http#http-request
+	pwResult, err := c.spGraphClient.Applications().ByApplicationId(id).AddPassword().Post(ctx, pwCredentialRequestBody, nil)
 	if err != nil {
 		return "", "", err
 	}
 
-	return *app.AppID, password, nil
+	return *appResult.GetAppId(), *pwResult.GetSecretText(), nil
 }
 
 func (c *Cluster) createServicePrincipal(ctx context.Context, appID string) (string, error) {
-	var sp azgraphrbac.ServicePrincipal
+	var result msgraph_models.ServicePrincipalable
```

Comment:
uh why is this named that way?  `ServicePrincipalable`

Reply:
I know it comes from MSFT, but :shrug:

Reply:
They generate all their Go interfaces to be named "thing-able" resulting in non-sense like this.